service/frontend/workflowHandler.go

```diff
@@ -2176,22 +1827,38 @@ func (wh *WorkflowHandler) GetWorkflowExecutionHistory(
 		token.IsWorkflowRunning = isWorkflowRunning
 		token.PersistenceToken = nil
 	}
+	rawHistoryQueryEnabled := wh.config.EnableRawHistoryQuery() && wh.config.EnableRawHistoryQueryByDomain(getRequest.GetDomain())
 
 	history := &gen.History{}
+	historyBlob := []*gen.DataBlob{}
 	history.Events = []*gen.HistoryEvent{}
 	if isCloseEventOnly {
 		if !isWorkflowRunning {
-			history, _, err = wh.getHistory(
-				scope,
-				domainID,
-				*execution,
-				lastFirstEventID,
-				nextEventID,
-				getRequest.GetMaximumPageSize(),
-				nil,
-				token.TransientDecision,
-				token.BranchToken,
-			)
+			if rawHistoryQueryEnabled {
+				historyBlob, _, err = wh.getRawHistory(
+					scope,
+					domainID,
+					*execution,
+					lastFirstEventID,
+					nextEventID,
+					getRequest.GetMaximumPageSize(),
+					nil,
+					token.TransientDecision,
+					token.BranchToken,
+				)
+			} else {
+				history, _, err = wh.getHistory(
+					scope,
+					domainID,
+					*execution,
+					lastFirstEventID,
+					nextEventID,
+					getRequest.GetMaximumPageSize(),
+					nil,
+					token.TransientDecision,
+					token.BranchToken,
+				)
+			}
 			if err != nil {
 				return nil, wh.error(err, scope)
 			}
```

Comment:
Also, check line 1866, `history.Events = history.Events[len(history.Events)-1 : len(history.Events)]`. This won't be valid. You need to handle this for both case.

Reply:
> Also, check line 1866, `history.Events = history.Events[len(history.Events)-1 : len(history.Events)]`. This won't be valid. You need to handle this for both case.

Done